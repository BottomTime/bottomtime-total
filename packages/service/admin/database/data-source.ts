import path from 'path';
import { DataSource } from 'typeorm';

export async function getDataSource(
  url: string,
  requireSsl: boolean,
): Promise<DataSource> {
  const client = new DataSource({
    type: 'postgres',
    url,
    entities: [path.resolve(__dirname, '../../src/data/**/*.entity.ts')],
    ssl: requireSsl,
  });
  return await client.initialize();
}
