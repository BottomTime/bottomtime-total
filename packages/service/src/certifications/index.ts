export * from './agency';
export * from './agencies.service';

export * from './certification';
export * from './certifications.controller';
export * from './certifications.service';
export * from './certifications.module';
