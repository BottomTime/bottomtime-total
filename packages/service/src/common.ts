import { z } from 'zod';
import { DepthUnit } from './constants';

export const BooleanString = z
  .enum(['true', 'false'])
  .transform((value) => value === 'true');

export const DepthSchema = z.object({
  depth: z.number().min(0),
  unit: z.nativeEnum(DepthUnit),
});
export type Depth = z.infer<typeof DepthSchema>;

export const GpsCoordinatesSchema = z.object({
  lat: z.number().gte(-90).lte(90),
  lon: z.number().gte(-180).lte(180),
});
export type GpsCoordinates = z.infer<typeof GpsCoordinatesSchema>;

export interface Range {
  min: number;
  max: number;
}
