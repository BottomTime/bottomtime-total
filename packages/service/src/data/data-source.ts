import path from 'path';
import { DataSource, DataSourceOptions } from 'typeorm';

import { Config } from '../config';

export const PostgresDataSourceOptions: DataSourceOptions = {
  type: 'postgres',
  url: Config.postgresUri,
  entities: [path.join(__dirname, './**/*.entity.ts')],
  synchronize: false,
};

export async function initDataSource(): Promise<DataSource> {
  const AppDataSource = new DataSource(PostgresDataSourceOptions);
  return await AppDataSource.initialize();
}
