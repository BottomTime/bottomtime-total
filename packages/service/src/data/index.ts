export * from './data-source';
export * from './get-query-vector';

export * from './certification.entity';
export * from './dive-site-review.entity';
export * from './dive-site.entity';
export * from './friend-request.entity';
export * from './friendship.entity';
export * from './tank.entity';
export * from './user-certification.entity';
export * from './user-oauth.entity';
export * from './user.entity';
