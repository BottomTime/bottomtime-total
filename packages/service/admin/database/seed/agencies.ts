import { AgencyEntity } from '../../../src/data';

export const SeedAgencies: AgencyEntity[] = [
  {
    id: '5ab222fa-3a71-4dbb-908a-92f281c16705',
    name: 'PADI',
    longName: 'Professional Association of Diving Instructors',
    logo: '/img/agencies/padi.png',
    website: 'https://www.padi.com',
    ordinal: 1,
  },
  {
    id: '7db437c8-4336-4a7b-b5b8-778f3c69905d',
    name: 'SSI',
    longName: 'Scuba Schools International',
    logo: '/img/agencies/ssi.png',
    website: 'https://www.divessi.com',
    ordinal: 2,
  },
  {
    id: '384d9fcb-6050-4760-a441-abc3e22d7e6e',
    name: 'CMAS',
    longName: 'Confédération Mondiale des Activités Subaquatiques',
    logo: '/img/agencies/cmas.png',
    website: 'https://www.cmas.org',
    ordinal: 3,
  },
  {
    id: '4967e7e8-dae1-43f8-88d5-f26aa273aba2',
    name: 'NAUI',
    longName: 'National Association of Underwater Instructors',
    logo: '/img/agencies/naui.png',
    website: 'https://www.naui.org',
    ordinal: 4,
  },
  {
    id: '04b4cd16-fceb-4692-8acf-cf2d15fac0b5',
    name: 'BSAC',
    longName: 'British Sub-Aqua Club',
    logo: '/img/agencies/bsac.png',
    website: 'https://www.bsac.com',
    ordinal: 5,
  },
  {
    id: '0f409465-5d1e-46c6-afcb-ab68843b7ec0',
    name: 'SDI',
    longName: 'Scuba Diving International',
    logo: '/img/agencies/sdi.png',
    website: 'https://www.tdisdi.com',
    ordinal: 6,
  },
  {
    id: '8acbfb2d-9546-4c2a-82b5-31b675fdcef8',
    name: 'RAID',
    longName: 'Rebreather Association of International Divers',
    logo: '/img/agencies/raid.png',
    website: 'https://www.diveraid.com',
    ordinal: 7,
  },
  {
    id: 'fc1087d3-b58e-4664-9f1c-899a93439432',
    name: 'TDI',
    longName: 'Technical Diving International',
    logo: '/img/agencies/tdi.png',
    website: 'https://www.tdisdi.com',
    ordinal: 8,
  },
];
