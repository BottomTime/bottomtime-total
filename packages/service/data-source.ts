import path from 'path';
import { DataSource } from 'typeorm';

import { PostgresDataSourceOptions } from './src/data/data-source';

export default new DataSource({
  ...PostgresDataSourceOptions,
  migrations: [path.join(__dirname, './migrations/*.ts')],
});
