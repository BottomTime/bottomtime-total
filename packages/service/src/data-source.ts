import 'reflect-metadata';
import { DataSourceOptions } from 'typeorm';

import { Config } from './config';
import * as Entities from './data';

export const PostgresDataSourceOptions: DataSourceOptions = {
  type: 'postgres',
  url: Config.postgresUri,
  entities: Object.values(Entities),
  synchronize: false,
};
