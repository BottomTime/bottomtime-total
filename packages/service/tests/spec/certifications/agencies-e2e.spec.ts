import { HttpServer, INestApplication } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';

import request from 'supertest';
import { Repository } from 'typeorm';

import { AgenciesService } from '../../../src/certifications';
import { AgenciesController } from '../../../src/certifications/agencies.controller';
import { AgencyEntity } from '../../../src/data';
import { dataSource } from '../../data-source';
import { TestAgencies } from '../../fixtures/agencies';
import { createTestApp } from '../../utils';

function getUrl(agencyId?: string): string {
  const url = '/api/agencies';
  return agencyId ? `${url}/${agencyId}` : url;
}

describe('Agencies E2E tests', () => {
  let Agencies: Repository<AgencyEntity>;
  let app: INestApplication;
  let server: HttpServer;

  beforeAll(async () => {
    Agencies = dataSource.getRepository(AgencyEntity);
    app = await createTestApp({
      imports: [TypeOrmModule.forFeature([AgencyEntity])],
      providers: [AgenciesService],
      controllers: [AgenciesController],
    });
    await app.init();
    server = app.getHttpServer();
  });

  beforeEach(async () => {
    await Agencies.save(TestAgencies);
  });

  afterAll(async () => {
    await app.close();
  });

  it('will retrieve the list of agencies', async () => {
    const { body } = await request(server).get(getUrl()).expect(200);
    expect(body).toMatchSnapshot();
  });

  it('will retrieve a single agency by its id', async () => {
    const { body } = await request(server)
      .get(getUrl(TestAgencies[1].id))
      .expect(200);
    expect(body).toMatchSnapshot();
  });

  it('will return a 404 response if the agency id does not exist', async () => {
    await request(server)
      .get(getUrl('f5d2198f-d173-4b49-b180-ea39b57e39ca'))
      .expect(404);
  });
});
