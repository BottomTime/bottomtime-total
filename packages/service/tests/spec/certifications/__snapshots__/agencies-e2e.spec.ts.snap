// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agencies E2E tests will retrieve a single agency by its id 1`] = `
{
  "id": "d85b955b-04f3-4803-ba74-c31c27bc26de",
  "logo": "https://www.naui.org",
  "name": "NAUI",
  "website": "https://www.naui.org",
}
`;

exports[`Agencies E2E tests will retrieve the list of agencies 1`] = `
{
  "data": [
    {
      "id": "aa170980-9a68-4168-9859-1f71e0d023df",
      "logo": "https://www.padi.com",
      "name": "PADI",
      "website": "https://www.padi.com",
    },
    {
      "id": "d85b955b-04f3-4803-ba74-c31c27bc26de",
      "logo": "https://www.naui.org",
      "name": "NAUI",
      "website": "https://www.naui.org",
    },
    {
      "id": "a6a87695-02fb-4caf-b5c2-6af5afbb0ff8",
      "logo": "https://www.divessi.com",
      "name": "SSI",
      "website": "https://www.divessi.com",
    },
  ],
  "totalCount": 3,
}
`;
